bench/ockam-rs/optimized/39zu1ift3djm93l6.ll

```diff
@@ -863,10 +863,13 @@ define hidden noundef ptr @_ZN5tokio7runtime2io6driver6Handle17deregister_source
   %7 = tail call noundef i64 @_ZN4core4sync6atomic11atomic_load17hd53d73d879a28131E.llvm.5777560861739151222(ptr noundef nonnull @_ZN3log20MAX_LOG_LEVEL_FILTER17h48f96c40e9c80142E, i8 noundef 0), !noalias !45
   %8 = icmp ult i64 %7, 6
   tail call void @llvm.assume(i1 %8)
-  %.not.i = icmp eq i64 %7, 5
-  br i1 %.not.i, label %.critedge11.i, label %_ZN3mio4poll8Registry10deregister17h8db01216e80c6d9cE.exit
+  %.0.i12.i = tail call noundef i8 @llvm.ucmp.i8.i64(i64 5, i64 %7)
+  switch i8 %.0.i12.i, label %_ZN3mio4poll8Registry10deregister17h8db01216e80c6d9cE.exit [
+    i8 0, label %.critedge11.i
+    i8 -1, label %.critedge11.i
+  ]
```

Comment:
Regression.

Reply:
I think this one would be handled by adding ConstantRange support for ucmp, so we see the result can only be 0 or -1 and eliminate the default case.